src/object_accessor.hpp

```diff
@@ -85,7 +85,24 @@ void Object::set_property_value_impl(ContextType& ctx, const Property &property,
         return;
     }
 
-    switch (property.type) {
+    if (is_array(property.type)) {
+        if (property.type == PropertyType::LinkingObjects)
+            throw ReadOnlyPropertyException(m_object_schema->name, property.name);
+        REALM_ASSERT(property.type == PropertyType::Object);
+
+        List list(m_realm, m_row.get_linklist(col));
+        list.remove_all();
```

Comment:
This appears to suffer from the same issue that you fixed in realm/realm-cocoa#5127. It looks like a preexisting issue, so it's not necessary to fix it prior to merging unless it's trivial to do so.